java/code/src/com/suse/manager/webui/controllers/maintenance/MaintenanceScheduleController.java

```diff
@@ -0,0 +1,252 @@
+import static com.suse.manager.webui.utils.SparkApplicationHelper.withCsrfToken;
+import static com.suse.manager.webui.utils.SparkApplicationHelper.withUser;
+import static com.suse.manager.webui.utils.SparkApplicationHelper.withUserPreferences;
+import static spark.Spark.delete;
+import static spark.Spark.get;
+import static spark.Spark.post;
+
+import com.redhat.rhn.common.localization.LocalizationService;
+import com.redhat.rhn.domain.role.RoleFactory;
+import com.redhat.rhn.domain.user.User;
+import com.redhat.rhn.manager.EntityExistsException;
+import com.redhat.rhn.manager.EntityNotExistsException;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.suse.manager.maintenance.IcalUtils;
+import com.suse.manager.maintenance.MaintenanceManager;
+import com.suse.manager.maintenance.rescheduling.RescheduleResult;
+import com.suse.manager.maintenance.rescheduling.RescheduleStrategy;
+import com.suse.manager.maintenance.rescheduling.RescheduleStrategyType;
+import com.suse.manager.model.maintenance.MaintenanceCalendar;
+import com.suse.manager.model.maintenance.MaintenanceSchedule;
+import com.suse.manager.reactor.utils.LocalDateTimeISOAdapter;
+import com.suse.manager.reactor.utils.OptionalTypeAdapterFactory;
+import com.suse.manager.webui.utils.gson.MaintenanceScheduleJson;
+import com.suse.manager.webui.utils.gson.ResultJson;
+import org.apache.http.HttpStatus;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import spark.ModelAndView;
+import spark.Request;
+import spark.Response;
+import spark.Spark;
+import spark.template.jade.JadeTemplateEngine;
+
+/**
+ * Controller class providing the backend for API calls to work with maintenance schedules.
+ */
+public class MaintenanceScheduleController {
+
+    private static IcalUtils icalUtils = new IcalUtils();
+    private static final MaintenanceManager MM = new MaintenanceManager();
+    private static final LocalizationService LOCAL = LocalizationService.getInstance();
+    private static final Gson GSON = new GsonBuilder()
+            .registerTypeAdapter(LocalDateTime.class, new LocalDateTimeISOAdapter())
+            .registerTypeAdapterFactory(new OptionalTypeAdapterFactory())
+            .serializeNulls()
+            .create();
+
+    private MaintenanceScheduleController() { }
+
+    /**
+     * Initialize routes for MaintenanceSchedule Api.
+     *
+     * @param jade the template engine
+     */
+    public static void initRoutes(JadeTemplateEngine jade) {
+        get("/manager/schedule/maintenance/schedules",
+                withUserPreferences(withCsrfToken(withUser(MaintenanceScheduleController::maintenanceSchedules))),
+                jade);
+        get("/manager/api/maintenance/schedule/list",
+                withUser(MaintenanceScheduleController::listSchedules));
```

Comment:
Suggestion: one of the advantages of having separated controllers for `Schedule` and `Calendar` is, that we could omit the "smurf" words (point 21 in https://blog.codinghorror.com/new-programming-jargon/). So we could simplify the methods to `list`, `getDetails/details`, `save`, `delete`. Same for the `MaintCalendarController`.

Reply:
Right I didn't think about those names after spitting the Controllers, but this makes total sense. Consider it done ;)